hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/deployment/ResourceIteratorTest.java

```diff
@@ -0,0 +1,116 @@
+
+    @Test
+    public void test_allEmptyFiles() throws IOException {
+        doTest(3, i -> "");
+    }
+
+    @Test
+    public void test_firstEmptyFile() throws IOException {
+        doTest(3, i -> i == 0 ? "" : "contents" + i);
+    }
+
+    @Test
+    public void test_middleEmptyFile() throws IOException {
+        doTest(3, i -> i == 1 ? "" : "contents" + i);
+    }
+
+    @Test
+    public void test_lastEmptyFile() throws IOException {
+        doTest(3, i -> i == 2 ? "" : "contents" + i);
+    }
+
+    private void doTest(int numFiles, IntFunction<String> contentsFactory) throws IOException {
+        // create files and their contents
+        Path directory = Files.createTempDirectory(ResourceIteratorTest.class.getSimpleName());
+        directory.toFile().deleteOnExit();
+
+        Path[] files = new Path[numFiles];
+        ResourceConfig[] configs = new ResourceConfig[numFiles];
+        String[] contents = new String[numFiles];
+
+        for (int i = 0; i < numFiles; i++) {
+            files[i] = directory.resolve("file" + i);
+            contents[i] = contentsFactory.apply(i);
+            try (Writer writer = Files.newBufferedWriter(files[i])) {
+                writer.append(contents[i]);
+            }
+            files[i].toFile().deleteOnExit();
+            configs[i] = new ResourceConfig(files[i].toUri().toURL(), String.valueOf(i), null);
+        }
+
+        // create the ResourceIterator
+        int partSize = 5;
+        ResourceIterator ri = new ResourceIterator(new LinkedHashSet<>(Arrays.asList(configs)), partSize);
+
+        // iterate it and check, that the contents match
+        int lastIndex = -1;
+        int lastOffset = 0;
+        while (ri.hasNext()) {
+            ResourcePart rp = ri.next();
+            int fileIndex = Integer.parseInt(rp.getDescriptor().getId());
+            assertTrue("part is too big", rp.getBytes().length <= partSize);
+            if (contents[fileIndex].length() > 0) {
+                assertTrue("zero-length part for non-empty file", rp.getBytes().length > 0);
+            }
+            assertTrue("some config was skipped", lastIndex == fileIndex || lastIndex + 1 == fileIndex);
+
+            if (fileIndex > lastIndex) {
+                if (lastIndex >= 0) {
+                    assertEquals("config not fully read", contents[lastIndex].length(), lastOffset);
+                }
+                lastIndex = fileIndex;
+                lastOffset = 0;
+            }
+            assertEquals("offsets not in sequence", lastOffset, rp.getOffset());
+            lastOffset += rp.getBytes().length;
+            assertEquals("contents don't match", contents[fileIndex].substring(rp.getOffset(), lastOffset), new String(rp.getBytes()));
+        }
+        assertEquals("not all config files read", lastIndex, numFiles - 1);
+        assertEquals("config not fully read", contents[lastIndex].length(), lastOffset);
+
+        // delete the files
+        for (Path file : files) {
```

Comment:
it's like early delete. If you run all the tests, they will be deleted earlier. I moved it to finally and remove deleteOnExit